runtime/jcl/common/java_dyn_methodhandle.c

```diff
@@ -95,9 +95,8 @@ lookupInterfaceMethod(J9VMThread *currentThread, J9Class *lookupClass, J9UTF8 *n
 			vmFuncs->setCurrentExceptionNLS(currentThread, J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR, J9NLS_JCL_PRIVATE_INTERFACE_REQUIRES_INVOKESPECIAL);
 			method = NULL;
 		} else {
-			*methodIndex = getITableIndexForMethod(method, lookupClass);
-			if (*methodIndex == -1) {
-				method = NULL;
+			if (J9_ARE_ANY_BITS_SET(J9_CLASS_FROM_METHOD(method)->romClass->modifiers, J9_JAVA_INTERFACE)) {
+				*methodIndex = getITableIndexForMethod(method, lookupClass);
```

Comment:
We should still be setting method to null if getITableIndexForMethod() returned -1.

Reply:
Also, is there a path through this code (or the caller) that allows returning with a null J9Method without having set an exception?

Reply:
Updated to return null and throw an exception.

lookupImpl() throws an exception when it returns null.
getMethodForSpecialSend() asserts notNull on the result